src/main/scala/ru/org/codingteam/keter/scenes/inventory/InventoryViewModel.scala

```diff
@@ -33,20 +34,31 @@ class InventoryViewModel(var inventory: Inventory) {
 
   val currentFolderItems = new ItemsViewModel[EquipmentItem](toVectorMap(inventory.allEquipment.toSeq)) {
 
+    def canEquip(item: EquipmentItem): Boolean = {
+      val inventoryProvidedCapabilities = inventory.body.foldLeft(mutable.MultiSet[Capability]())(_ ++= _.provides)
+      val inventoryRequiredCapabilities = inventory.equipment.foldLeft(mutable.MultiSet[Capability]())(_ ++= _.requires)
+
+      inventoryProvidedCapabilities --= inventoryRequiredCapabilities
+      (item.requires & inventoryProvidedCapabilities.toSet) == item.requires
+    }
+
     def equipped(item: EquipmentItem): Boolean = {
       inventory.equipment.contains(item)
     }
 
-    def toggleSelected() = {
+    def toggleSelected(): Unit = {
       selectedItem foreach { item =>
         var backpack = inventory.backpack
         var equipment = inventory.equipment
+
         if (equipped(item)) {
           equipment -= item
           backpack += item
-        } else {
+        } else if (canEquip(item)) {
           backpack -= item
           equipment += item
+        } else {
+          return
```

Comment:
I suggest to return the missing capabilities from the `canEquip` method, log them here, and add a `// TODO: Show missing capabilities on screen`.

Reply:
@ForNeVeR ok